generate_report_ui.gs

```diff
@@ -0,0 +1,263 @@
+   }
+  else {
+    reportType.addItem("Most Executed Script", "mostExecutedScript", true)
+    reportType.addItem("Most Active Users", "mostActiveUsers", false)
+    reportType.addItem("Owners of Scripts", "ownerOfScripts", false);
+  }
+  reportType.setOnChangeAction(CardService.newAction()
+        .setFunctionName("reportTypeCallback"));
+  reportTypesection.addWidget(reportType);
+  
+  if(!e.formInput || e.formInput.reportType != "ownerOfScripts") {
+  var limitResponse = CardService.newTextInput()
+    .setFieldName("limitResponse")
+    .setTitle("Enter the Number Of Responses ");
+    reportTypesection.addWidget(limitResponse);
+  }
+  card.addSection(reportTypesection);
+  
+  var section2 = CardService.newCardSection();
+  // Create text Paragraph for placing url of report
+  if(reportUrlVal) {
+    var reportUrl = CardService.newTextParagraph()
+    .setText("<b>Report Url</b>\n <a href='"+reportUrlVal+"'>"+reportUrlVal+"</a>");
+    section2.addWidget(reportUrl);
+  }
+    
+  // Create a button for generating report
+  var getReportAction = CardService.newAction()
+      .setFunctionName('generateReport'); 
+  var getReport = CardService.newTextButton()
+      .setText('Generate Report')
+      .setOnClickAction(getReportAction)
+      .setTextButtonStyle(CardService.TextButtonStyle.FILLED);
+  section2.addWidget(getReport);
+  card.addSection(section2);
+  
+   return card.build();
+}
+
+/**
+* Callback function on clicking the GET REPORT button and create a card with most executed script and most active users
+* @return {CardService.Card} The card to show to the user
+*/
+function generateReport(e) {
+  var title, projTitle;
+  var toTime = new Date();
+  var fromTime;
+  var millisInBetween;
+  var input  = e.formInput;
+  var projectFilter = input.projectFilter;
+  var reportType = input.reportType;
+  var limitResponse = input.limitResponse;
+  if(!limitResponse) limitResponse = 1;
+    
+  if(input.timeFilter == "LAST_HOUR") { 
+    title = "Last Hour";
+    millisInBetween = 60*60*1000;
+  } else if(input.timeFilter == "LAST_6_HOUR") {
+    title = "Last 6 Hours";
+    millisInBetween = 6*60*60*1000;
+  } else if(input.timeFilter == "LAST_24_HOUR") {
+    title = "Last 24 Hours";
+    millisInBetween = 24*60*60*1000;
+  } else if(input.timeFilter == "LAST_7_DAYS") {
+    title = "Last 7 Days";
+    millisInBetween = 7*24*60*60*1000;
+  } else {
+    title = "Last 30 Days";
+    millisInBetween = 30*24*60*60*1000;
+  }
+  
+  if(projectFilter == "SPECIFIC_PROJECT") {
+    projTitle = "Specific Project";
+  } else if(projectFilter == "CUSTOM_PROJECT") {
+    projTitle = "Custom Projects";
+  } else if(projectFilter == "SYSTEM_PROJECT") {
+    projTitle = "System Projects";
+  } else {
+    projTitle = "All Projects";
+  }
+  
+  // variables to get the start time and end time
+  var fromTime = new Date(toTime.getTime() - millisInBetween);
```

Comment:
In the same function there are too many declaration of fromTime. Please fix that. One variable should be declared once. Similarly below

Reply:
Updated